### DISCUSSION TITLE
Sponsors chapters indicators for locally downloaded videos?

### DISCUSSION THREAD

**Discussion by xfzv** [tomasklaen/uosc#261] [Q&A] [answered]
I'm using [po5/mpv_sponsorblock/](https://github.com/po5/mpv_sponsorblock/). While the sponsors chapters indicators are working fine when watching directly from YouTube, they're not when playing the same video locally, despite the chapters being properly detected in both cases. I'm watching/downloading with [yt-dlp](https://github.com/yt-dlp/yt-dlp). Relevant download options:

```
--embed-chapters
--output "%(uploader)s - %(upload_date)s - %(title)s [%(id)s].%(ext)s"
--parse-metadata "%(title)s:%(meta_title)s"
--parse-metadata "%(uploader)s:%(meta_artist)s"
--sponsorblock-mark "all"
```

- Directly from YouTube:
![image](https://user-images.githubusercontent.com/78810647/192110083-be579adc-74af-4425-921c-013e69148ba4.png)

- Locally:
![image](https://user-images.githubusercontent.com/78810647/192110087-94803eec-6ac6-467e-b4db-7b0b32b2ccd2.png)

I tried to add `local_pattern=" %[([%a%d%-_]+)].[mw][kpe][v4b]m?$"` to `~/.config/mpv/sponsorblock.conf` but it doesn't make any difference.

Is there a way to also get the red indicators for locally downloaded videos?

Thanks!

**Comment by christoph-heinrich** [comment, marked as answer]
I don't think sponsorblock works for local videos. What are the chapters called in the local file that you want to get recognized as sponsor spots?

**Reply by xfzv** [reply]
`--sponsorblock-mark "all"` does include the sponsor chapters when downloading the videos, although the names and the timestamps are somewhat different when watching directly.

- Directly from YouTube:
![image](https://user-images.githubusercontent.com/78810647/192112074-337b7efd-ca33-4601-8f48-1c57c86f9f0c.png)

- Local video:
![image](https://user-images.githubusercontent.com/78810647/192112084-a9bbd520-2ac3-4585-9b17-4075ce27529b.png)

Not related but thank you for [mpv-quality-menu](https://github.com/christoph-heinrich/mpv-quality-menu), I've been using it for quite a while now, it's awesome. Very nice integration with uosc too.

**Reply by christoph-heinrich** [reply]
That seems like something we can fix.

**Reply by christoph-heinrich** [reply]
Can you please test #262?

**Reply by xfzv** [reply]
![image](https://user-images.githubusercontent.com/78810647/192116341-c2a9c1a5-14a4-4ce7-a536-7f2b9033a114.png)

The sponsor sections are now properly highlighted, however the respective chapters indicators are gone? :disguised_face:

**Reply by christoph-heinrich** [reply]
That is because a sponsor range is next to it, and chapters that enclose a range are not shown.
If you have suggestions for alternative behavior you'll have to talk that over with @tomasklaen.

**Reply by tomasklaen** [reply]
Hiding chapter indicators made sense in the past when they were drawn on timeline as lines or dots, so them colliding with the range was pointless, ugly, and confusing. But since chapter indicator is now a diamond on top of the timeline, I would actually prefer drawing them now.

**Reply by xfzv** [reply]
Thank you, perfect after updating to latest commits:

![image](https://user-images.githubusercontent.com/78810647/192135879-8cae77dd-aa4c-40d0-a6f0-b3adee2fc3c9.png)
